src/docfx/lib/log/Telemetry.cs

```diff
@@ -17,47 +18,75 @@ internal static class Telemetry
     // https://github.com/microsoft/ApplicationInsights-Home/blob/master/EndpointSpecs/Schemas/Bond/EventData.bond#L19
     private const int MaxEventPropertyLength = 8192;
     private const int MaxChildrenLength = 5;
-    private static readonly TelemetryClient s_telemetryClient = new(TelemetryConfiguration.CreateDefault());
+
+    private static TelemetryClient TelemetryClient { get; }
 
     // Set value per dimension limit to int.MaxValue
     // https://github.com/microsoft/ApplicationInsights-dotnet/issues/1496
     private static readonly MetricConfiguration s_metricConfiguration = new(int.MaxValue, int.MaxValue, new MetricSeriesConfigurationForMeasurement(false));
 
-    private static readonly Metric s_operationStartMetric =
-        s_telemetryClient.GetMetric(
+    private static Metric OperationStartMetric { get; }
+
+    private static Metric OperationEndMetric { get; }
+
+    private static Metric ErrorCountMetric { get; }
+
+    private static Metric FileLogCountMetric { get; }
+
+    private static Metric BuildFileTypeCountMetric { get; }
+
+    private static Metric GithubRateLimitMetric { get; }
+
+    private static Metric MarkdownElementCountMetric { get; }
+
+    private static Metric HtmlElementCountMetric { get; }
+
+    private static readonly string s_version =
+        typeof(Telemetry).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "<null>";
+
+    private static readonly string s_os = RuntimeInformation.OSDescription ?? "<null>";
+    private static readonly string s_correlationId = EnvironmentVariable.CorrelationId ?? Guid.NewGuid().ToString("N");
+    private static readonly string s_sessionId = EnvironmentVariable.SessionId ?? "<null>";
+    private static readonly AsyncLocal<bool> s_isRealTimeBuild = new();
+
+    private static string s_repo = "<null>";
+    private static string s_branch = "<null>";
+
+    static Telemetry()
+    {
+        using var telemetryConfiguration = TelemetryConfiguration.CreateDefault();
+        using var dependencyTrackingTelemetryModule = new DependencyTrackingTelemetryModule();
+        dependencyTrackingTelemetryModule.Initialize(telemetryConfiguration);
```

Comment:
We'll need to sanitize SAS from Azure storage URL. Please check sample code: https://dev.azure.com/ceapex/Engineering/_git/Docs.Build?path=/src/shared/Telemetry.cs&version=GBdevelop&line=21&lineEnd=22&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Reply:
Added